fix: DIA-2097: [FE] Create Project modal closes automatically when user is at Home Page

nick-skriabin opened the pull request:
Fixes the issue when the project creation modal closes automatically on the second open attempt.

This pull request includes several changes to the `CreateProject` and `HomePage` components in the `web/apps/labelstudio/src/pages` directory. The changes focus on improving the handling of project creation and deletion, as well as enhancing navigation behaviors.

Improvements to project creation and deletion:

* [`web/apps/labelstudio/src/pages/CreateProject/CreateProject.jsx`](diffhunk://#diff-37799f8bcc3b80b87f49ceb8ec48563ba527a6597bb0a6cf9d4e5fd6550d836cL172-R173): Refactored the `onDelete` callback to use an internal `performClose` function to handle asynchronous operations more cleanly. [[1]](diffhunk://#diff-37799f8bcc3b80b87f49ceb8ec48563ba527a6597bb0a6cf9d4e5fd6550d836cL172-R173) [[2]](diffhunk://#diff-37799f8bcc3b80b87f49ceb8ec48563ba527a6597bb0a6cf9d4e5fd6550d836cR184-R185)

Enhancements to navigation and modal handling:

* [`web/apps/labelstudio/src/pages/Home/HomePage.tsx`](diffhunk://#diff-5beb2cd206dd6c5d34a094f966f0b306e197713ce75e3d11d4b55a1af603f4efR12): Added `useHistory` import and utilized `history.replace` to prevent the modal from auto-closing after project creation. [[1]](diffhunk://#diff-5beb2cd206dd6c5d34a094f966f0b306e197713ce75e3d11d4b55a1af603f4efR12) [[2]](diffhunk://#diff-5beb2cd206dd6c5d34a094f966f0b306e197713ce75e3d11d4b55a1af603f4efR56) [[3]](diffhunk://#diff-5beb2cd206dd6c5d34a094f966f0b306e197713ce75e3d11d4b55a1af603f4efL172-R182)

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Prevent project creation modal from auto closing

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14081947390)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2097/create-project-auto-closes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14081947390

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14081947390

nick-skriabin committed:
Fix project creation modal auto closing

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14100756573)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2097/create-project-auto-closes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14100756573

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14100756573

nick-skriabin committed:
Remove unused dependency

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14124031628)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2097/create-project-auto-closes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14124031628

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14124031628
